File: python/longestCommonPrefix.py
arr = ["flower", "flight", "flow"]

def lcp(arr):
    size = len(arr)
    if size == 0: return ""
    if size == 1: return []
    
    arr.sort()
    
    start = 0
    end = min(len(arr[0]), len(arr[size-1]))
    
    while start < end and arr[0][start] == arr[size-1][start]:
        start += 1

    string = arr[0][:start]
    return string    

res = lcp(arr)
print(res)

File: python/longestCommonPrefix.js
const strs = ["flow", "flower","flight"]
 
const lcp = (strs) => {
	strs = strs.sort()
  const size = strs.length
  let start = 0, end = Math.min(strs[0].length, strs[size-1].length)
  
  while(start < end && strs[0][start] == strs[size-1][start]){
  	start += 1
  }
  return strs[0].substring(0, start)
}

let res = lcp(strs)
console.log(res)